File: solutions/7/7.java
class Main {
    public static void fillArray(int[] is_prime, int size, int value) {
        for (int i = 0; i < size; i++) {
            is_prime[i] = value;
        }
    }

    public static int nthPrime(int position) {
        int max_size = 1000000;

        int[] is_prime = new int[max_size];

        fillArray(is_prime, max_size, 1);

        int i, j, prime_position = 0;
        for (i = 2; i < max_size; i++) {
            if (is_prime[i] == 1) {
                prime_position += 1;

                if (prime_position == position) {
                    return i;
                }

                for (j = i * 2; j < max_size; j += i) {
                    is_prime[j] = 0;
                }
            }
        }

    return -1;
}

    public static void main(String args[]) {
        int result = nthPrime(10001);

        System.out.println(result);
    }
}

File: solutions/7/7.rs
fn nth_prime(position: usize) -> usize {
    const MAX_SIZE: usize = 1000000;

    let mut is_prime: [usize; MAX_SIZE] = [1; MAX_SIZE];

    let mut prime_position: usize = 0;
    for i in 2..MAX_SIZE {
        if is_prime[i] == 1 {
            prime_position += 1;

            if prime_position == position {
                return i;
            }

            for j in (i * 2..MAX_SIZE).step_by(i) {
                is_prime[j] = 0;
            }
        }
    }

    return 0;
}

fn main() {
    let result: usize = nth_prime(10001);

    println!("{}", result);
}
